packaging/windows-setup/installer.iss

```diff
@@ -60,8 +60,67 @@ Type: filesandordirs; Name: "{app}\jre"; Components: java
 Type: filesandordirs; Name: "{app}\jre"; Components: java
 
 [Code]
+#include "lib/JsonHelpers.pas"
+const
+CONFIGURL = 'https://download.nodecdn.net/containers/atl/launcher/json/config.json';
+FALLBACKx86 = 'https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.9%2B9.1/OpenJDK17U-jre_x86-32_windows_hotspot_17.0.9_9.zip';
+FALLBACKx64 = 'https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.9%2B9.1/OpenJDK17U-jre_x64_windows_hotspot_17.0.9_9.zip';
+
 var
   DownloadPage: TDownloadWizardPage;
+  FallbackUrl, FallbackHash, Url, Hash, Folder: WideString;
+
+procedure GetJreInfo;
+  var
+    WinHttpReq: Variant;
+    Json, OS: string;
+    JsonParser: TJsonParser;
+    JsonRoot, BundledJreObject, OSObject: TJsonObject;
+begin
+  if IsWin64 then 
+    begin
+      OS := 'windowsx64'
+      FallbackUrl := FALLBACKx64
+      FallbackHash := '6c491d6f8c28c6f451f08110a30348696a04b009f8c58592191046e0fab1477b'
+    end
+  else
+    begin
+      OS := 'windowsx86'
+      FallbackUrl := FALLBACKx86
+      FallbackHash := '2f9fe8b587400e89cd3ef33b71e0517ab99a12a5ee623382cbe9f5078bf2b435'
+    end;
+  Try
+    WinHttpReq := CreateOleObject('WinHttp.WinHttpRequest.5.1');
+    WinHttpReq.Open('GET', CONFIGURL, False);
+    WinHttpReq.Send('');
+    if WinHttpReq.Status = 200 then
+     begin
+        Json := WinHttpReq.ResponseText
+        if ParseJsonAndLogErrors(JsonParser, Json) then
+          begin
+            JsonRoot := GetJsonRoot(JsonParser.Output);
+            if not FindJsonObject(JsonParser.Output, JsonRoot, 'bundledJre', BundledJreObject) or
+            not FindJsonObject(JsonParser.Output, BundledJreObject, OS, OSObject) or
+            not FindJsonString(JsonParser.Output, OSObject, 'url', Url) or
+            not FindJsonString(JsonParser.Output, OSObject, 'hash', Hash) or
+            not FindJsonString(JsonParser.Output, OSObject, 'folder',Folder) then
+            begin
+              RaiseException('Failed to read from ' + CONFIGURL + ', falling back to defaults')
```

Comment:
This is the first time I've ever written Pascal, didn't think of a better way to handle these two error conditions but perhaps this repeated RaiseException could be simplified to one.